src/polyswarm_api/resources.py

```diff
@@ -34,13 +34,37 @@ class Engine(core.BaseJsonResource):
 
     def __init__(self, content, api=None):
         super(Engine, self).__init__(content=content, api=api)
-        self.address = content['address'].lower()
-        self.name = content.get('name')
+        self.id = str(content['id'])
+        self.name = content['name']
+
+        if 'address' in content and isinstance(content['address'], str):
+            self.address = content['address'].lower()
+        else:
+            self.address = None
+
+        self.engine_type = content.get('engineType')
+        self.verified = content.get('status') == 'verified'
+
+        # These fields can be `null`; don't replace w/ default value in `get()`
+        self.artifact_types = tuple(content.get('artifactTypes') or ())
```

Comment:
There is no need for converting everything to `tuple`, it is just unnecessry overhead. These are being parsed from json and are being converted to `list`. It makes the code harder to read and creates other objects that we don't need. To keep type consistency, plz change to 
```python
self.artifact_types = content.get('artifactTypes') or []
```
Also applies to all the other instances where this happens.